src/engine/enginemaster.cpp

```diff
@@ -97,6 +100,13 @@ EngineMaster::EngineMaster(UserSettingsPointer pConfig,
 
     m_pMasterDelay = new EngineDelay(group, ConfigKey(group, "delay"));
     m_pHeadDelay = new EngineDelay(group, ConfigKey(group, "headDelay"));
+    m_pInputLatencyCompensationDelay = new EngineDelay(group,
+        ConfigKey(group, "inputLatency"));
+    m_pInputLatencyCompensationHeadphonesDelay = new EngineDelay(group,
+        ConfigKey(group, "headMixDelay"));
```

Comment:
We do need this because the headphones need to have the same latency compensation applied as the master mix. Otherwise the headphones and master mix will be out of time even if the user is right next to the master speakers. A separate EngineDelay instance is required for every delayed buffer because each EngineDelay keeps its own internal buffer. I can add a comment here explaining that.